src/Val.cc

```diff
@@ -3491,26 +3487,28 @@ ValPtr TypeVal::DoClone(CloneState* state)
 	return {NewRef{}, this};
 	}
 
-VectorVal::VectorVal(VectorTypePtr t) : VectorVal(t, new vector<std::optional<ZVal>>()) { }
-
-VectorVal::VectorVal(VectorTypePtr t, std::vector<std::optional<ZVal>>* vals) : Val(t)
+VectorVal::VectorVal(VectorTypePtr t) : Val(t)
 	{
-	vector_val = vals;
 	yield_type = t->Yield();
 
 	auto y_tag = yield_type->Tag();
 	any_yield = (y_tag == TYPE_VOID || y_tag == TYPE_ANY);
 	managed_yield = ZVal::IsManagedType(yield_type);
 	}
 
+VectorVal::VectorVal(VectorTypePtr t, std::vector<std::optional<ZVal>>* vals) : VectorVal(t)
+	{
+	vector_val = std::move(*vals);
```

Comment:
The previous API accepted `nullptr`.
```suggestion
	if ( vals )
		vector_val = std::move(*vals);
```